### PR TITLE
Expand dns alias security considerations

### LINKED ISSUE #2487: Extend next-hop alias security considerations to cover trust in DNS resolution
Without going through a proxy, clients implement CNAME cloaking mitigations on the basis of DNS information they see during resolution, which is totally reasonable. If resolution is subverted (or whatever) and the CNAME information is modified, these client checks might not work. When going through a proxy, the proxy is the one doing DNS resolution, so it seems like any sort of funny business that one might do to DNS answers for the proxy might affect the client's ability to implement cloaking mitigations correctly. The security considerations might benefit from a bit of expansion to cover this additional aspect of the threat model.

### PR DESCRIPTION
**Pull request by tfpauly**
Closes #2487

### CODE REVIEW

**Comment by tfpauly** [comment]
@chris-wood please take a look!

File: draft-ietf-httpbis-alias-proxy-status.md [hunk 1]
```diff
@@ -155,10 +155,18 @@ An implementation MAY include incomplete information in the `next-hop-aliases` p
...
-trusts its proxy to provide accurate information. This information is intended to be used as
+trusts the proxy to provide accurate information. This information is intended to be used as
 a hint, and SHOULD NOT be used for making security decisions about the identity of a resource accessed
 through the proxy.
 
+Inspecting CNAME chains can be used to detect cloaking of trackers or malicious hosts. However, the
+CNAME records could be omitted by a recursive resolver that is trying to hide this form of cloaking.
+Such attacks to avoid cloaking detection apply equally to clients directly performing DNS name
+resolution and proxies performing DNS name resolution on behalf of client. A malicious proxy could
+also choose to not report these CNAME chains in order to hide the cloaking. In general, client can
```
Comment (chris-wood) [inline draft-ietf-httpbis-alias-proxy-status.md:166]:
```suggestion
also choose to not report these CNAME chains in order to hide the cloaking. In general, clients can
```

File: draft-ietf-httpbis-alias-proxy-status.md [hunk 2]
```diff
@@ -155,10 +155,18 @@ An implementation MAY include incomplete information in the `next-hop-aliases` p
 
 The `next-hop-aliases` parameter does not include any DNSSEC information or imply that DNSSEC was used.
 The information included in the parameter can only be trusted to be valid insofar as the client
-trusts its proxy to provide accurate information. This information is intended to be used as
+trusts the proxy to provide accurate information. This information is intended to be used as
 a hint, and SHOULD NOT be used for making security decisions about the identity of a resource accessed
 through the proxy.
 
+Inspecting CNAME chains can be used to detect cloaking of trackers or malicious hosts. However, the
+CNAME records could be omitted by a recursive resolver that is trying to hide this form of cloaking.
```
Comment (chris-wood) [inline draft-ietf-httpbis-alias-proxy-status.md:163]:
```suggestion
CNAME records could be omitted by a recursive or authoritative resolver that is trying to hide this form of cloaking.
```

File: draft-ietf-httpbis-alias-proxy-status.md [hunk 3]
```diff
@@ -155,10 +155,18 @@ An implementation MAY include incomplete information in the `next-hop-aliases` p
...
+Such attacks to avoid cloaking detection apply equally to clients directly performing DNS name
+resolution and proxies performing DNS name resolution on behalf of client. A malicious proxy could
```
Comment (chris-wood) [inline draft-ietf-httpbis-alias-proxy-status.md:165]:
```suggestion
In particular, recursive or authoritative resolvers can omit these records for both clients directly performing DNS name
resolution and proxies performing DNS name resolution on behalf of client. A malicious proxy could
```

**Review by chris-wood — APPROVED** [review]

**Pull request merged**